### PR TITLE
Remove a couple bits of Swift 3 compatibility code from type resolution.

### PR DESCRIPTION
**Pull request by rudkx**

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Please smoke test

File: lib/Sema/TypeCheckType.cpp [hunk 1]
```diff
@@ -2784,109 +2779,9 @@ Type TypeResolver::resolveTupleType(TupleTypeRepr *repr,
...
-  }
-  assert(!Postfixes.empty() && isa<IdentTypeRepr>(LastType));
-
-  // Now, we know we can fix-it. do it.
-  SmallVector<TypeRepr *, 4> Protocols(Types.begin(), Types.end() - 1);
-  Protocols.push_back(LastType);
-
-  // Emit fix-it to enclose composition part into parentheses.
-  TypeRepr *InnerMost = Postfixes.back();
-  TC.diagnose(InnerMost->getLoc(), diag::protocol_composition_with_postfix,
```
Comment (slavapestov) [inline lib/Sema/TypeCheckType.cpp]: Is this diagnostic now dead?

Comment (rudkx) [reply lib/Sema/TypeCheckType.cpp]: Yup, just pushed another commit to remove it. Thanks.

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Comment by rudkx** [comment]
@swift-ci Please test source compatibility

**Pull request merged**